librz/core/cbin.c

```diff
@@ -1695,113 +1726,116 @@ static int bin_relocs(RzCore *r, PJ *pj, int mode, int va) {
 		rz_table_set_columnsf(table, "XXss", "vaddr", "paddr", "type", "name");
 	} else if (IS_MODE_JSON(mode)) {
 		pj_a(pj);
-	} else if (IS_MODE_SET(mode)) {
-		rz_flag_space_set(r->flags, RZ_FLAGS_FS_RELOCS);
 	}
 
-	rz_rbtree_foreach (relocs, iter, reloc, RzBinReloc, vrb) {
-		ut64 addr = rva(r->bin, reloc->paddr, reloc->vaddr, va);
-		if (IS_MODE_SET(mode) && (is_section_reloc(reloc) || is_file_reloc(reloc))) {
-			/*
-			 * Skip section reloc because they will have their own flag.
-			 * Skip also file reloc because not useful for now.
-			 */
-		} else if (IS_MODE_SET(mode)) {
-			set_bin_relocs(r, reloc, addr, &db, &sdb_module);
-			add_metadata(r, reloc, addr, mode);
-		} else if (IS_MODE_SIMPLE(mode)) {
-			rz_cons_printf("0x%08" PFMT64x "  %s\n", addr, reloc->import ? reloc->import->name : "");
-		} else if (IS_MODE_RZCMD(mode)) {
-			char *name = reloc->import
-				? strdup(reloc->import->name)
-				: (reloc->symbol ? strdup(reloc->symbol->name) : NULL);
-			if (name && bin_demangle) {
-				char *mn = rz_bin_demangle(r->bin->cur, NULL, name, addr, keep_lib);
-				if (mn) {
+	RBNode *relocs = NULL;
+	if (r->bin->cur && r->bin->cur->o) {
```

Comment:
if possible without too much hassle, please extract all the inner code in a separate function because having such a long `if` is painful for my eyes :D